examples/md-flexible/tests/ThermostatTest.cpp

```diff
@@ -111,13 +117,45 @@ TEST_F(ThermostatTest, MultiComponentTest) {
     EXPECT_NEAR(temperature, targetTemperature1, 0.1);
   }
 
+  // Check no particles have zero temperature (ensuring brownian motion has been applied).
+  // We also want to later compare against the current velocity. For simplicity's sake, we use the oldF buffer of the
+  // particle to do so.
+  for (auto &particle : _autopas) {
+    for (size_t dim = 0; dim < 3; ++dim) {
+      // Check some velocity applied.
+      EXPECT_THAT(particle.getV()[dim], ::testing::Not(::testing::DoubleNear(0, 1e-12)));
+    }
+    // set the oldF to the velocity
+    particle.setOldF(particle.getV());
+  }
+
   // set system to a different temperature through apply and check that the temperature matches for each component
   constexpr double targetTemperature2 = targetTemperature1 + 2.;
+
+  // calculate the expected scaling factors
+  // Note that different components have different scaling factors
+  std::vector<double> scalingFactors{};
+  scalingFactors.reserve(temperatureMap.size());
+  for (size_t i = 0; i < temperatureMap.size(); ++i) {
+    scalingFactors[i] = std::sqrt(targetTemperature2 / temperatureMap[i]);
+  }
+
+  // apply thermostat
   Thermostat::apply(_autopas, _particlePropertiesLibrary, targetTemperature2, std::numeric_limits<double>::max());
+
+  // check each component's temperature is adjusted to targetTemperature2
   temperatureMap = Thermostat::calcTemperatureComponent(_autopas, _particlePropertiesLibrary);
   for (auto &[typeId, temperature] : temperatureMap) {
     EXPECT_NEAR(temperature, targetTemperature2, 1e-9);
   }
+
+  // Check that every particle's velocity has been scaled correctly
+  for (auto iter = _autopas.begin(); iter.isValid(); ++iter) {
```

Comment:
could also be range-based as far as I can tell

Reply:
Done in `7eef7c9e`